### PR TITLE
fix cppcheck complains

### PR DESCRIPTION
**Pull request by utoni**
Signed-off-by: Toni Uhlig <matzeton@googlemail.com>

### CODE REVIEW

File: systemd-sslh-generator.c [hunk 1]
```diff
@@ -55,9 +55,13 @@ static int get_listen_from_conf(const char *filename, char **listen[]) {
                             config_setting_source_line(addr));
                     return -1;
                 } else {
+                    char *resolved_listen;
+
                     (*listen)[i] = malloc(strlen(resolve_listen(hostname, port)));
```
Comment (yrutschle) [inline systemd-sslh-generator.c:60]: doesn't `resolve_listen()` leak here as well?

Comment (yrutschle) [reply systemd-sslh-generator.c:60]: (basically it seems to me you want to assign `resolved_listen`  once, then free it.)

Comment (utoni) [reply systemd-sslh-generator.c:60]: true, overlooked that one

**Comment by yrutschle** [comment]
Merged in less than a year, hurray!

Thanks for the fixes.

**Pull request merged**